rewrote Configuring Lookback section (#111)
logdna/logdna-agent-v2: docs/README.md

* [Running as Non-Root on Kubernetes](KUBERNETES.md#run-as-non-root)
* [Running as Non-Root on OpenShift](OPENSHIFT.md#run-as-non-root)

> If you configure the agent to run as non-root, review the [documentation about enabling "statefulness" for the agent](KUBERNETES.md#enabling-persistent-agent-state).

Reviewer: `KUBERNETES.md#enabling-persistent-agent-state` points to a location that doesn't exist in the markdown file. Was the idea to also modify the `KUBERNETES.md` file?

Reviewer: oh, I see that it relates to #109 , nvm.

> If you configure the agent to run as non-root, review the [documentation about enabling "statefulness" for the agent](KUBERNETES.md#enabling-persistent-agent-state).

Reviewer 2: ```suggestion
If you configure the LogDNA Agent to run as non-root, review the [documentation about enabling "statefulness" for the agent](KUBERNETES.md#enabling-persistent-agent-state).
```

### Configuring Lookback

> The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK`.

Reviewer: ```suggestion
The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK` variable.
```

The valid values for this option are:

>    * When set to **`none`**: lookback is disabled, and LogDNA continues to read from where the file ended at the time of the stop/restart action. 

Reviewer: In technical terms `none` means the Agent will go to the end of the file, ignoring previously written lines, and it will read new lines of the file as those are added (tailing).

```suggestion
   * When set to **`none`**: lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were wrote before the time the Agent restarted.
```

Author: Ah, thanks much ore clear. Two grammar changes though please:

"... Agent will start reading new..." 
"...lines that were written before..."

   * When set to **`smallfiles`** (default):
       * If there is information in the “state file”, use the last recorded state. 
       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end. 

>    * When set to **`start`**:

Reviewer: hm... in the github markdown viewer, this is shown as `start: *` ...

Author: fixedit,thanks

Reviewer: There's still an issue with indentation of the list, that is shown in the viewer:

<img width="580" alt="Screenshot 2021-03-31 at 09 40 27" src="https://user-images.githubusercontent.com/2931196/113108490-55636080-9205-11eb-8f2f-8386cd4c93e7.png">

    * If there is information in the “state file”, use the last recorded state. 
    * If the file is not present in the “state file”, start at the beginning. 

> The state file location is defined using the LOGDNA_DB_PATH env variable (the default path is /var/lib/logdna/agent_state.db).

Reviewer: ```suggestion
The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is /var/lib/logdna/agent_state.db).
```

**Notes:**
* If you configure the agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the agent.

> * When upgrading from LogDNA Agent versin 3.0 to 3.1, the state file will initially be empty, so the lookback setting will be used for existing files. After that (i.e. on process restart), the state file will be present and will be used.

Reviewer: version :)

Author: oh my gosh my horrible typing! Thank you for catching that. I will fix and add to the PR.

### Configuring Lookback

The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by the `LOGDNA_LOOKBACK` variable.

The valid values for this option are:
   * When set to **`none`**:

>       * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.

Reviewer 2: ```suggestion
      * lookback is disabled, and LogDNA Agent will read new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.
```

>       * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.

Reviewer 2: ```suggestion
      * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the LogDNA Agent restarted.
```

> By default, the agent provides a "stateful", or persistent, collection of files that can be referenced whenever the agent is restarted, in order to return (or look back) to see any files that were ingested during the time that the agent was not running. The state file location is defined using the `LOGDNA_DB_PATH` environment variable in the YAML file (the default path is /var/lib/logdna/agent_state.db).

Reviewer: `LOGDNA_DB_PATH` setting should point to a directory.

```suggestion
By default, the agent provides a "stateful", or persistent, collection of files that can be referenced whenever the agent is restarted, in order to return (or look back) to see any files that were ingested during the time that the agent was not running. The state directory location is defined using the `LOGDNA_DB_PATH` environment variable in the YAML file (the default path is `/var/lib/logdna`).
```

   * When set to **`smallfiles`** (default):
       * If there is information in the “state file”, use the last recorded state. 
       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end. 
   * When set to **`start`**:

>     * If there is information in the “state file”, use the last recorded state. 
>     * If the file is not present in the “state file”, start at the beginning. 

Reviewer: I think the indentation was messing up the sublist.

```suggestion
       * If there is information in the “state file”, use the last recorded state. 
       * If the file is not present in the “state file”, start at the beginning. 
```

Author: Hmmm let me try again, becuse I need those to be level-two bullets, so mor eindented.

Reviewer: This issue persists after the latest changes.

> The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is /var/lib/logdna/agent_state.db).

Reviewer 2: ```suggestion
The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is `/var/lib/logdna/agent_state.db`).
```

**Notes:**

> * If you configure the agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the agent.

Reviewer 2: ```suggestion
* If you configure the LogDNA Agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the LogDNA Agent.
```
